docs/sources/installation/istio.md

```diff
@@ -0,0 +1,179 @@
+and
+https://grpc.github.io/grpc/core/md_doc_load-balancing.html
+3. Set the `appProtocol` of `grpclb` to `tcp`
+4. Set `publishNotReadyAddresses` to `true`
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: loki-query-frontend
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-query-frontend
+spec:
+  ports:
+  - appProtocol: http
+    name: http
+    port: 3100
+    protocol: TCP
+    targetPort: http
+  - appProtocol: tcp
+    name: grpclb
+    port: 9095
+    protocol: TCP
+    targetPort: grpc
+  publishNotReadyAddresses: true
+  selector:
+    app: loki-query-frontend
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-query-frontend
+  type: ClusterIP
+```
+
+### Querier service
+
+Set the `appProtocol` of the `grpc` service to `tcp`
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: loki-querier
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-querier
+  name: loki-querier
+  namespace: observability
+spec:
+  ports:
+  - appProtocol: http
+    name: http
+    port: 3100
+    protocol: TCP
+    targetPort: http
+  - appProtocol: tcp
+    name: grpc
+    port: 9095
+    protocol: TCP
+    targetPort: grpc
+  selector:
+    app: loki-querier
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-querier
+  type: ClusterIP
+
+```
+
+### Ingester Service and Ingester Headless Service
+
+Set the `appProtocol` of the `grpc` port to `tcp` 
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: loki-ingester-(headless)
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-ingester
+  name: loki-ingester-headless
+spec:  
+  clusterIP: None (if headless)
+  ports:
+  - name: http
+    port: 3100
+    protocol: TCP
+    targetPort: http
+  - appProtocol: tcp
+    name: grpc
+    port: 9095
+    protocol: TCP
+    targetPort: grpc
+  selector:
+    app: loki-ingester
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-ingester
+  type: ClusterIP
+```
+
+### Distributor Service
+
+Set the `appProtocol` of the `grpc` port to `tcp` 
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: loki-distributor
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-distributor
+spec:
+  ports:
+  - name: http
+    port: 3100
+    protocol: TCP
+    targetPort: http
+  - name: grpc
+    port: 9095
+    protocol: TCP
+    targetPort: grpc
+    appProtocol: tcp
+  selector:
+    app: loki-distributor
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-distributor
+  sessionAffinity: None
+  type: ClusterIP
+
+```
+
+## Memberlist Service
```

Comment:
Not sure why this is ## and not ### level like the previous headings?

Make the following modifications to the file for the Memberlist service.